Framework/src/InfrastructureSpecReader.cxx

```diff
@@ -402,4 +407,19 @@ T InfrastructureSpecReader::readSpecEntry(const std::string&, const boost::prope
   throw std::runtime_error("Unknown entry type: " + std::string(typeid(T).name()));
 }
 
+void InfrastructureSpecReader::validateName(const std::string& name)
+{
+  std::string const trunc = name.substr(0, 16);
+  auto test = sDeviceNamesCache.emplace(trunc);
+  if (!test.second) { // the second of the pair indicates if there was an insertion, i.e. it was not already there
+    std::string errorMessage = std::string("Multiple devices share the same first 16 characters of their names: ") + trunc + "[" + name.substr(16) + "]. Probable cause: name was longer than 16 characters and was truncated. Disable one of the faulty tasks/aggregator/postprocessing or call the detector expert to change the name.";
```

Comment:
Again, devices do not have the 16-character limitation, `DataDescription` has. This is very misleading.